### PR TITLE
update version variables for v1.4

### PR DESCRIPTION
**Pull request by shannonbradshaw**
Per instructions here, filing this in preparation for the 1.4 release:
https://github.com/kubeflow/kubeflow/blob/master/docs_dev/releasing.md#updating-version-numbers-etc-for-the-upcoming-release-major-minor-or-patch

Steps 4-6 of those instructions seem to be very out of date. I'll follow with an issue to address.

### CODE REVIEW

**Review by Arhell — COMMENTED** [review]
/lgtm

File: layouts/shortcodes/kf-latest-version.html [hunk 1]
```diff
@@ -1 +1 @@
-v1.3.0
\ No newline at end of file
+v1.4.0
```
Comment (zijianjoy) [inline layouts/shortcodes/kf-latest-version.html:1]: Should this file also be updated in master branch?

**Comment by zijianjoy** [comment]
/approve

Thank you for helping to cut v1.4 branch!

**Pull request merged**